src/init.cpp

```diff
@@ -1933,6 +1933,17 @@ bool AppInitMain()
         // Link thread groups
         LinkAutoSpendThreadGroup(&threadGroupAutoSpend);
 
+        // Validate wallet mining address.
+        std::string sAddress = gArgs.GetArg("-miningaddress", "");
+        if (!sAddress.empty()) {
+            CTxDestination dest = DecodeDestination(sAddress);
+
+            auto pt = GetMainWallet();
+            if (pt && pt->IsMine(dest) != ISMINE_SPENDABLE) {
+                return InitError(strprintf(_("Invalid -miningaddress: '%s' not owned by wallet"), sAddress));
```

Comment:
This has an issue; as it hangs up the wallet and doesn't exit:
```
2021-04-20T23:24:24Z ThreadStakeMiner: starting
2021-04-20T23:24:24Z Veil Miner started
2021-04-20T23:24:24Z Error: Invalid -miningaddress: 'bv1q7cmcjcazyxger9y9llnqpuhj33pmrn9hl0ny24' not owned by wallet
2021-04-20T23:24:24Z tor: Thread interrupt
2021-04-20T23:24:24Z torcontrol thread exit
2021-04-20T23:24:24Z opencon thread exit
2021-04-20T23:24:24Z dnsseed thread exit
2021-04-20T23:24:24Z Shutdown: In progress...
2021-04-20T23:24:24Z addcon thread exit
2021-04-20T23:24:24Z msghand thread exit
2021-04-20T23:24:24Z net thread exit
2021-04-20T23:24:24Z Imported mempool transactions from disk: 3 succeeded, 0 failed, 0 expired, 0 already there
2021-04-20T23:24:37Z ThreadStakeMiner: interrupted
2021-04-20T23:24:37Z Veil Miner started
```
```
./veil-cli stop
error: Could not connect to the server 127.0.0.1:58812

Make sure the veild server is running and that you are connecting to the correct RPC port.
```

Reply:
This appears to be an existing issue. I can reproduce it here:
https://github.com/Veil-Project/veil/blob/02cc068b03abe4aeac551762f24ba621819ad733/src/init.cpp#L1972-L1974

Let me file a separate issue...

Reply:
I filed PR #935 with a fix.